pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -586,13 +587,17 @@ func (c *Cluster) getOSDInfo(d *appsv1.Deployment) (OSDInfo, error) {
 	}
 
 	locationFound := false
-	for _, a := range container.Args {
+	for _, a := range container.Command {
 		locationPrefix := "--crush-location="
-		if strings.HasPrefix(a, locationPrefix) {
+		if strings.Contains(a, locationPrefix) {
 			locationFound = true
 			// Extract the same CRUSH location as originally determined by the OSD prepare pod
 			// by cutting off the prefix: --crush-location=
-			osd.Location = a[len(locationPrefix):]
+			rx := regexp.MustCompile(`--crush-location="(.+?)"`)
```

Comment:
Could you factor this regex match snippet out to a helper method and then add a unit test? String matching is always a perfect candidate for unit tests

Reply:
Moved to a separate method with unit tests.